kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/data/vector/DefaultGenericVector.java

```diff
@@ -136,6 +137,17 @@ public MapValue getMap(int rowId) {
         return (MapValue) values[rowId];
     }
 
+    @Override
```

Comment:
Can `DefaultConstantVector` be derived from `DefaultGenericVector`? If yes, then we can reduce the code size bit.

Reply:
We could combine them and use a function like "value accessor" possibly? Then all the class does really is cast

Reply:
Can't restrict the types then though

Reply:
You can add the type checks in `DefaultConstantVector` constructor (this will just have the constructor)

Reply:
Combined them here instead